File: Hardeep3pm/FirstTime/app/src/main/java/com/example/firsttime/SecondActivity.java
package com.example.firsttime;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.ContextMenu;
import android.view.MenuItem;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.ListView;
import android.widget.TextView;
import android.widget.Toast;

public class SecondActivity extends AppCompatActivity {
TextView textView;
String value_email;
    Button movebtn;
    ListView listView;
    String[] countrylist = {"India","USA","Australia","America","Delhi","Japan","china"};
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);
        value_email = getIntent().getStringExtra("first");

      
  listView = findViewById(R.id.listview);
  ArrayAdapter<String> arrayAdapter = new ArrayAdapter<String>(this, android.R.layout.simple_spinner_item,countrylist);
  arrayAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
  listView.setAdapter(arrayAdapter);

 registerForContextMenu(listView);
 listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
     @Override
     public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
         Toast.makeText(SecondActivity.this, "clicke "+countrylist[position], Toast.LENGTH_SHORT).show();
     }
 });
        movebtn = findViewById(R.id.move_to_third);
    textView = findViewById(R.id.value_textview);
    textView.setText(value_email);
        movebtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(SecondActivity.this,ThirdActivity.class);
                startActivity(intent);
            }
        });
    }

    @Override
    public void onCreateContextMenu(ContextMenu menu, View v, ContextMenu.ContextMenuInfo menuInfo) {
        super.onCreateContextMenu(menu, v, menuInfo);
        getMenuInflater().inflate(R.menu.context_menu,menu);
    }

    @Override
    public boolean onContextItemSelected(@NonNull MenuItem item) {

        int id = item.getItemId();
        if(id == R.id.contactusid)
        {
            Toast.makeText(this, "clicked "+item.getTitle(), Toast.LENGTH_SHORT).show();
        }else if(id == R.id.treeid)
        {
            Toast.makeText(this, "clicked "+item.getTitle(), Toast.LENGTH_SHORT).show();
        }
        return super.onContextItemSelected(item);

    }
}

File: kushalapp/BasickComponent/app/src/main/java/com/cbitss/basickcomponent/SecondActivity.kt
package com.cbitss.basickcomponent


import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.ContextMenu
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.*

class SecondActivity : AppCompatActivity() {
    lateinit var  listview:ListView
    var text="some"
    var numberlist = arrayOf("324324","34234234","667675","3453453","242342","45345345","235345345","4534534","324324","34234234","667675","3453453","242342","45345345","235345345","4534534","324324","34234234","667675","3453453","242342","45345345","235345345","4534534","324324","34234234","667675","3453453","242342","45345345","235345345","4534534","324324","34234234","667675","3453453","242342","45345345","235345345","4534534","324324","34234234","667675","3453453","242342","45345345","235345345","4534534")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)
        listview = findViewById(R.id.listview)

        var listadapter = ArrayAdapter<String>(applicationContext,R.layout.spinner_layout,numberlist)
        listview.adapter = listadapter

      listview.setOnItemClickListener(object : AdapterView.OnItemClickListener{
          override fun onItemClick(p0: AdapterView<*>?, p1: View?, postion: Int, p3: Long) {
              text = numberlist.get(postion)
              var popupmenuexaple = PopupMenu(this@SecondActivity,listview)
              popupmenuexaple.menuInflater.inflate(R.menu.contextmenufile,popupmenuexaple.menu)
              popupmenuexaple.show()
              popupmenuexaple.setOnMenuItemClickListener {
                  val id = it.itemId
                  when(id)
                  {
                      R.id.logutbtn ->{
                          Toast.makeText(applicationContext,"Log Out Clicked",Toast.LENGTH_SHORT).show()
                      }
                      R.id.updateprofitle ->{
                          Toast.makeText(applicationContext,"Update Clicked"+text,Toast.LENGTH_SHORT).show()
                      }
                      R.id.aboutbtn ->{
                          val cardintent = Intent(this@SecondActivity,CardVIewExample::class.java)
                          startActivity(cardintent)
                      }
                  }
                  true
              }

          }
      })

        registerForContextMenu(listview)

    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.optionmenu,menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {

        val id = item.itemId
        when(id)
        {
            R.id.logutbtn ->{
                Toast.makeText(applicationContext,"Log Out Clicked",Toast.LENGTH_SHORT).show()
            }
            R.id.updateprofitle ->{
                Toast.makeText(applicationContext,"selected value "+text,Toast.LENGTH_SHORT).show()
            }
            R.id.aboutbtn ->{
               val cardintent = Intent(this@SecondActivity,CardVIewExample::class.java)
                startActivity(cardintent)
            }
        }
        return true
    }

    override fun onCreateContextMenu(menu: ContextMenu?, v: View?, menuInfo: ContextMenu.ContextMenuInfo?) {
        super.onCreateContextMenu(menu, v, menuInfo)
        menuInflater.inflate(R.menu.contextmenufile,menu)

    }

    override fun onContextItemSelected(item: MenuItem): Boolean {
        val id = item.itemId

        when(id)
        {
            R.id.logutbtn ->{
                Toast.makeText(applicationContext,"Log Out Clicked",Toast.LENGTH_SHORT).show()
            }
            R.id.updateprofitle ->{
                Toast.makeText(applicationContext,"Update Clicked"+text,Toast.LENGTH_SHORT).show()
            }
            R.id.aboutbtn ->{
                val cardintent = Intent(this@SecondActivity,CardVIewExample::class.java)
                startActivity(cardintent)
            }
        }
        return true

    }

}